src/Cache/CacheClient.php

```diff
@@ -163,7 +171,7 @@ public function deleteCache(string $cacheName): DeleteCacheResponse
      */
     public function set(string $cacheName, string $key, string $value, int $ttlSeconds = 0): SetResponse
     {
-        return $this->dataClientWrapper->getClient()->set($cacheName, $key, $value, $ttlSeconds);
+        return $this->getNextDataClient()->set($cacheName, $key, $value, $ttlSeconds);
```

Comment:
Seems like a fair approach, to just round robin between the clients. Should be reasonably fair, unless there is some kind of blockage on a connection, though for my use case, I'd jam everything in at the start, so there isn't much opportunity to weight choice towards connections with fewer streams/lower latency.

Reply:
yeah, round robin is the way we've been doing it in the other languages so far.